app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,48 @@
+class PaymentsController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+  load_resource :conference, find_by: :short_title
+  authorize_resource :conference_registrations, class: Registration
+
+  def index
+    @payments = current_user.payments
+  end
+
+  def new
+    @total_amount_to_pay = Ticket.total_price(@conference, current_user, paid: false)
+    @unpaid_ticket_purchases = current_user.ticket_purchases.unpaid.by_conference(@conference)
+  end
+
+  def create
+    @unpaid_ticket_purchases = current_user.ticket_purchases.unpaid.by_conference(@conference)
+    @total_amount_to_pay = Ticket.total_price(@conference, current_user, paid: false)
+
+    customer = Stripe::Customer.create(
+      :email => params[:stripeEmail],
+      :source  => params[:stripeToken]
+    )
+
+    gateway_response = Stripe::Charge.create(
+      :customer    => customer.id,
+      :amount      => @total_amount_to_pay.cents,
+      :description => 'Rails Stripe customer',
+      :currency    => @conference.tickets.first.price_currency
+    )
+
+    payment = Payment.purchase(gateway_response, current_user, @conference)
```

Comment:
Why not handle this the way you did it for active merchant?
You can build a Payment with all the necessary data, and then call `payment.purchase` (no parameters) which will call Stripe and handle all that.